libnetwork/sandbox.go

```diff
@@ -347,6 +348,15 @@ func (sb *Sandbox) GetEndpoint(id string) *Endpoint {
 	return nil
 }
 
+// RefreshResolverPolicy makes sure the resolver will only forward DNS requests to
+// upstream resolvers if proxying is enabled, and there's at least one endpoint with
+// external connectivity.
+func (sb *Sandbox) RefreshResolverPolicy() {
```

Comment:
We're setting up a trap for future code archaeologists (read: ourselves in a year) by including the `RefreshResolverPolicy` refactor—which is now _only_ relevant for Linux—in a commit titled "Windows DNS resolver forwarding," for which the only other change in this source file is the addition of a Windows-only struct field. I think this refactor should be split into a separate PR, or at least its own commit, for the benefit of our future selves.

Reply:
I've un-done that change.